### PR TITLE
Feature/home screen

### PR DESCRIPTION
**Pull request by abdallahFc**
#### Description

> This pull request to refactor home screen and add comment screen.

### CODE REVIEW

File: presentation/viewmodels/src/main/java/com/cupcake/viewmodels/comment/CommentViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            ::onGetCommentsPostSuccess,
+            ::onGetCommentPostFailure
+        )
+    }
+
+    private fun onGetCommentsPostSuccess(posts: List<Post>) {
+        _state.update {
+            it.copy(
+                isLoading = false,
+                error = null,
```
Comment (mustafa-sab3alarab) [inline presentation/viewmodels/src/main/java/com/cupcake/viewmodels/comment/CommentViewModel.kt:51]: why assign null to an error when it's already null

File: presentation/viewmodels/src/main/java/com/cupcake/viewmodels/posts/PublicPostsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            ::onGetPostsSuccess,
+            ::onGetPostsFailure
+        )
+    }
+
+    private fun onGetPostsSuccess(posts: List<Post>) {
+        _state.update {
+            it.copy(
+                isLoading = false,
+                isError = false,
```
Comment (mustafa-sab3alarab) [inline presentation/viewmodels/src/main/java/com/cupcake/viewmodels/posts/PublicPostsViewModel.kt:41]: the same here it's has a default value

**Review by msalarab — CHANGES_REQUESTED** [review]
well job

**Review by msalarab — APPROVED** [review]

**Review by msalarab — APPROVED** [review]

**Review by msalarab — APPROVED** [review]

File: data/repository/src/main/java/com/cupcake/repository/JobFinderRepositoryImpl.kt [hunk 3]
```diff
@@ -115,12 +118,25 @@ class JobFinderRepositoryImpl @Inject constructor(
         return wrapResponseWithErrorHandler { api.getPosts() }.map { it.toPost() }
     }
 
+
+    override suspend fun getFollowingPosts(): List<Post> {
+        val fakePosts = listOf(
```
Comment (TarekIdrees) [inline data/repository/src/main/java/com/cupcake/repository/JobFinderRepositoryImpl.kt:145]: is this fake data still here?

**Review by TarekIdrees — CHANGES_REQUESTED** [review]

**Pull request merged**